code/espurna/thinkspeak.ino

```diff
@@ -29,6 +30,10 @@ const char THINGSPEAK_REQUEST_TEMPLATE[] PROGMEM =
 bool _tspk_enabled = false;
 bool _tspk_clear = false;
 
+String _tspk_host;
+String _tspk_url;
+unsigned int _tspk_port;
```

Comment:
Just copy from `URL` below? If am not mistaken, c++ creates default copy constructors which will allow to have:
```
// assuming there is new URL(const String& host, uint16_t port, const String& path) constructor
URL one {"host", 80, "/path"};
URL two {"other", 81, "/other_path"};
one = two;
```

Reply:
...or, just always using `_tspk_url` without copying.